update/Pos/Update/Worker.hs

```diff
@@ -67,14 +69,21 @@ checkForUpdate = do
                 svNumber . upSoftwareVersion . cpsUpdateProposal
         let newestCPS =
                 maximumBy (comparing cpsToNumericVersion) confirmedProposals
-        logInfo $
-            sformat
-                ("There are new confirmed update proposals for our application: "
-                 %listJsonIndent 2%
-                 "\n The newest one is: "%build%" and we want to download it")
-                (cpsUpdateProposal <$> confirmedProposals)
-                (cpsUpdateProposal newestCPS)
-        downloadUpdate newestCPS
+
+        ifM (noServersProvided newestCPS) (logInfo "There is no servers provided") $ do
+            logInfo $
+                sformat
+                    ("There are new confirmed update proposals for our application: "
+                    %listJsonIndent 2%
+                    "\n The newest one is: "%build%" and we want to download it")
+                    (cpsUpdateProposal <$> confirmedProposals)
+                    (cpsUpdateProposal newestCPS)
+            downloadUpdate newestCPS
+
+    noServersProvided :: ConfirmedProposalState -> m Bool
+    noServersProvided ConfirmedProposalState {..} = 
+        null <$> views (lensOf @UpdateParams) upUpdateServers
```

Comment:
I do not understand this. `ConfirmedProposalState` is not used by this function. Why is it passed here?
I wrote it earlier in #2612, btw.

Reply:
Sorry, I just have forgotten about that :(